analysis/3_coregister-and-source-recon-bulk.py

```diff
@@ -0,0 +1,240 @@
+    polhemus_headshape = np.loadtxt(op.join(polhemus_dir, 'polhemus_headshape.txt'))
+    polhemus_nasion = np.loadtxt(op.join(polhemus_dir, 'polhemus_nasion.txt'))
+    polhemus_rpa = np.loadtxt(op.join(polhemus_dir, 'polhemus_rpa.txt'))
+    polhemus_lpa = np.loadtxt(op.join(polhemus_dir, 'polhemus_lpa.txt'))
+    
+    #  Get coreg filenames
+    filenames = source_recon.rhino.get_coreg_filenames(recon_dir, subject)
+
+    # Save
+    np.savetxt(filenames["polhemus_nasion_file"], polhemus_nasion)
+    np.savetxt(filenames["polhemus_rpa_file"], polhemus_rpa)
+    np.savetxt(filenames["polhemus_lpa_file"], polhemus_lpa)
+    np.savetxt(filenames["polhemus_headshape_file"], polhemus_headshape)
+
+############## ------- Directories ---------- ############
+
+# Directories
+DATA_DIR = coinsmeg.RAW_DIR # this is the same as BASE_DIR as raw data is stored in the base directory
+preproc_dir = coinsmeg.PREPROCESSED_DIR
+recon_dir = op.join(coinsmeg.DERIVATIVES_DIR, "recon")
+
+# Get subjects
+subs = []
+
+data_sub_folders = sorted(filter(lambda path: "sub-" in path, glob(DATA_DIR + '/*'))) # returns a list of paths e.g., '/ohba/pi/lhunt/datasets/coins-meg_data/sub-22'
+
+for subject in data_sub_folders:
+    subs.append(pathlib.Path(subject).stem) # subs is now a list of ['sub-01', 'sub-02', ...]
+
+# Create a list with '_run-X' appended for each subject
+sub_run_combos = []
+for sub in subs:
+    for run in range(1, 5):  # Loop from 1 to 4
+        sub_run_combos.append(f"{sub}_run-{run}") # sub_run_combos is now a list of ['sub-01_run-1', 'sub-01_run-2', ...]
+
+print(sub_run_combos)
+
+# test with only two sub/run combos
+sub_run_combos =['sub-04_run-1','sub-04_run-2']
+
+
+############## -------  Run coregistration and source reconstruction   ---------- ############
+
+for sub_run_combo in sub_run_combos:
+
+    # extract the subject and run id
+    subject_id, run_id = sub_run_combo.split('_') # will produce subject_id = 'sub-XX', run_id = 'run-X'
+    # set paths for where to look for necessary files
+    anat_dir = coinsmeg.get_sub_anat_dir(subject_id)
+    smri_file = f"{anat_dir}/{subject_id}_T1w.nii"
+    fif_file = f'{preproc_dir}/{subject_id}_ses-2-meg_task-coinsmeg_{run_id}_meg_transsss/{subject_id}_ses-2-meg_task-coinsmeg_{run_id}_meg_transsss_preproc_raw.fif'
+    
+    if not os.path.exists(smri_file):
+        print(f"WARNING: smri_file does not exist for {sub_run_combo}!")
+        continue # skip over the rest of the code for this sub_run_combo
+
+    source_recon.rhino.compute_surfaces(
+        smri_file,
+        recon_dir,
+        sub_run_combo,
+        include_nose=True,
+    )
+    # check in fsleyes
+    source_recon.rhino.surfaces_display(recon_dir, sub_run_combo)
+    
+    polhemus_dir = op.join(recon_dir, sub_run_combo, "polhemus")
+    # make directory if it doesn't yet exist
+    os.makedirs(polhemus_dir, exist_ok=True)
+
+    from osl.source_recon.rhino.polhemus import extract_polhemus_from_info
```

Comment:
Perhaps better to import at the top of the script? We shouldn't need to run the same import statement multiple times for each subject